lib/helpers/indexes/decorateDiscriminatorIndexOptions.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = function decorateDiscriminatorIndexOptions(schema, indexOptions) {
+  // If the model is a discriminator and has an index, add a
+  // partialFilterExpression by default so the index will only apply
+  // to that discriminator.
+  const discriminatorName = schema.discriminatorMapping && schema.discriminatorMapping.value;
```

Comment:
Opportunity for refactoring here: pass in `discriminatorKey` and `discriminatorValue` instead of `schema`